python/vyos/config.py

```diff
@@ -193,50 +194,56 @@ def show_config(self, path=[], default=None, effective=False):
         """
         return self._config_source.show_config(path, default, effective)
 
-    def get_cached_dict(self, effective=False):
-        cached = self._dict_cache.get(effective, {})
+    def get_cached_root_dict(self, effective=False, multi=False):
+        cached = self._dict_cache.get((effective, multi), {})
         if cached:
-            config_dict = cached
-        else:
-            config_dict = {}
+            return cached
 
+        config_dict = self._dict_cache.get((effective, False), None)
+        if config_dict is None:
             if effective:
-                if self._running_config:
-                    config_dict = json.loads((self._running_config).to_json())
+                config = self._running_config
+            else:
+                config = self._session_config
+
+            if config:
+                config_dict = json.loads(config.to_json())
             else:
-                if self._session_config:
-                    config_dict = json.loads((self._session_config).to_json())
+                config_dict = {}
 
-            self._dict_cache[effective] = config_dict
+        self._dict_cache[(effective, False)] = config_dict
+        if multi:
+            config_dict = vyos.xml.multi_to_list([], config_dict)
+            self._dict_cache[(effective, True)] = config_dict
 
         return config_dict
 
-    def get_config_dict(self, path=[], effective=False, key_mangling=None, get_first_key=False):
+    def get_config_dict(self, path=[], effective=False, key_mangling=None, get_first_key=False, multi=False):
```

Comment:
I'd like to see this as "always true" to avoid error by accident - also then the data model is always defined and strict.

Reply:
Me too but this should be done in another patch where you can make sure all works as should